docs/examples/general/data_loading/webdataset.ipynb

```diff
@@ -0,0 +1,514 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #29.        if not read_ahead: # if read_ahead is False the cycling is handled by the external generator
</pre><p>I don't understand this. If read_ahead is False, we use cycle generator, which basically gives you an infinite dataset, which might be not what the user wants. The user might still want to detect the end of an epoch. Could you explain a bit more how this is supposed to work?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/3153/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
<p>As per the reader specification, the reader should supply the data indefinitely (and currently it's not possible to expose the epoch_size)</p>

Reply:
<p>Alright, just for the future refetence, it turned out that the dataset needed to be represented as a generator function and the cycling is now being handled by the external_source</p>